### ISSUE TITLE
WebAssembly Calling Graph in B2C

### ISSUE THREAD

**Issue by RedBlue1776** [dotnet/AspNetCore.Docs#18287]
[EDIT by guardrex to add the topic metadata. btw @RedBlue1776 ... when you use the **This page** feedback button+form at the bottom of the topic, the metadata is automatically added, which also automatically pings the author of the topic.]

How do I call the Graph API from a WebAssembly?

I worked with these Azure B2C demos extensively, and they leverage Microsoft.Identity.Web instead of MSAL.

https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2

https://github.com/AzureAD/microsoft-identity-web

How would I implement the equivalent in WebAssembly? 
Which flow is most similar to what I would need to use? (Xamarin, Javascript, or Microsoft.Identity.Web)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bfb23446-3b01-73a8-70e2-697367065264
* Version Independent ID: 67b01234-eeaf-dcee-929f-6e2f6880e313
* Content: [Secure an ASP.NET Core Blazor WebAssembly standalone app with Azure Active Directory B2C](https://docs.microsoft.com/en-us/aspnet/core/security/blazor/webassembly/standalone-with-azure-active-directory-b2c?view=aspnetcore-3.1)
* Content Source: [aspnetcore/security/blazor/webassembly/standalone-with-azure-active-directory-b2c.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/security/blazor/webassembly/standalone-with-azure-active-directory-b2c.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @RedBlue1776 ... It might not be supported at this time. Ordinarily (for example in a non-B2C app), one would follow the guidance at ...

https://docs.microsoft.com/en-us/aspnet/core/security/blazor/webassembly/additional-scenarios?view=aspnetcore-3.1#request-additional-access-tokens

... and on the request, it would also be possible to configure the client with the scope via the guidance at ...

https://docs.microsoft.com/en-us/aspnet/core/security/blazor/webassembly/additional-scenarios?view=aspnetcore-3.1#configure-the-httpclient-handler

However, I was working up some coverage for B2C similar to ...

https://docs.microsoft.com/en-us/aspnet/core/security/blazor/webassembly/azure-active-directory-groups-and-roles?view=aspnetcore-3.1

... for getting Azure AD security groups and roles into a B2C app, where one has to make a B2C Graph API call, and it :boom:. I ran out of time to investigate further given that we're gearing up for the 3.2 release.

https://github.com/dotnet/AspNetCore.Docs/issues/17683#issuecomment-627498149

Now, keep in mind that my test app is actually trying to hit **_two APIs_**, one for the server API in a Hosted Blazor scearnio that returns weather data (that bit works with the default config) and a second **_additional scope_** for Graph API. The approach works in a non-B2C AAD app, but it :boom: in a B2C app.

Since this issue focuses specifically on the scenario, I'll add the metadata to your opening remark and work it as a slightly separate issue. Whatever the answer, it will either unblock my other issue or result in some kind of *unsupported NOTE* being added to the topic.

Stand-by ... these issues are normally triaged on Fridays.

**Comment by guardrex** [comment]
Since my tests are with a Hosted Blazor scenario, try it with the default scope and see if it works (i.e., only one API being hit, not two).

**Comment by guardrex** [comment]
One more thing @RedBlue1776 ..

> Microsoft.Identity.Web

That's an Azure Identity Platform v2 set of APIs. The bits here for WASM are on v1. That's being worked out right now by engineering. We'll get docs up for v2 (and new packages) as soon as they're available.

**Comment by guardrex** [comment]
@RedBlue1776 I don't have any further information on this at this time. See Javier's reaction on ...

https://github.com/dotnet/AspNetCore.Docs/issues/17683#issuecomment-630051326

I'll need to work with my test app a bit more in that case to see if I can work it out.

The current guidance is at ...

https://docs.microsoft.com/en-us/aspnet/core/security/blazor/webassembly/additional-scenarios?view=aspnetcore-3.1#request-additional-access-tokens

... which I know works for AAD. If you don't end up reaching a working solution, open an issue for engineering at ...

https://github.com/dotnet/aspnetcore/issues

... and they'll probably need to see your code, so it's probably best if you put up a repro project and link that into your issue.

Please add a `cc: @guardrex` to your issue so that I can follow the discussion.
